src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+            throw new DataProcessingException("Can't create manufacturer in DB. "
+                    + "Manufacturer: " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String sqlGetManufacturerRequest =
+                "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturedStatement =
+                        connection.prepareStatement(sqlGetManufacturerRequest)) {
+            getManufacturedStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturedStatement.executeQuery();
+            Manufacturer manufacturer = null;
+            if (resultSet.next()) {
+                manufacturer = getManufacturer(resultSet);
+            }
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get manufacturer from DB by id:" + id, e);
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String sqlUpdateManufacturerRequest =
+                "UPDATE manufacturers "
+                        + "SET name = ?, "
+                        + "country = ? "
+                        + "where id = ? "
+                        + "AND is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturersStatement =
+                        connection.prepareStatement(sqlUpdateManufacturerRequest)) {
+            updateManufacturersStatement.setString(1, manufacturer.getName());
+            updateManufacturersStatement.setString(2, manufacturer.getCountry());
+            updateManufacturersStatement.setLong(3, manufacturer.getId());
+            updateManufacturersStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer with this id. ID:"
+                    + manufacturer.getId(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String sqlDeleteManufacturerRequest =
+                "UPDATE manufacturers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturersStatement =
+                        connection.prepareStatement(sqlDeleteManufacturerRequest)) {
+            deleteManufacturersStatement.setLong(1, id);
+            return deleteManufacturersStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete manufacturer from DB. id:"
+                    + id, e);
+        }
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
```

Comment:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) {
```
I think it's a redundant parameter.